### PR TITLE
Add a default value for the docker.network configuration

### PR DESCRIPTION
**Pull request by jbdoumenjou**
### What does this PR do?

Add a default value for the traefik.docker.network configuration.

### Motivation

Implement the #3449 

### More

- [x] Added/updated tests
- [x] Added/updated documentation

### Additional Notes

I can't test it due to the limitation of the docker test library (we can't manipulate the networks).

### CODE REVIEW

File: docs/configuration/backends/docker.md [hunk 1]
```diff
@@ -71,6 +71,12 @@ usebindportip = true
 #
 swarmMode = false
 
+# Define a default docker network to use for connections to all containers.
```
Comment (emilevauge) [inline docs/configuration/backends/docker.md]: Maybe we should add the fact that it's possible to override this with the label `traefik `traefik.docker.network`

File: provider/docker/docker.go [hunk 2]
```diff
@@ -45,6 +45,7 @@ type Provider struct {
 	ExposedByDefault      bool             `description:"Expose containers by default" export:"true"`
 	UseBindPortIP         bool             `description:"Use the ip address from the bound port, rather than from the inner network" export:"true"`
 	SwarmMode             bool             `description:"Use Docker on Swarm Mode" export:"true"`
+	Network               string           `description:"Default Docker network used" export:"true"`
```
Comment (emilevauge) [inline provider/docker/docker.go]: Shouldn't we call it `DefaultNetwork` instead?

Comment (ldez) [reply provider/docker/docker.go]: To be homogeneous with `domain`, we need to keep `Network`.

File: provider/docker/docker.go [hunk 3]
```diff
@@ -45,6 +45,7 @@ type Provider struct {
 	ExposedByDefault      bool             `description:"Expose containers by default" export:"true"`
 	UseBindPortIP         bool             `description:"Use the ip address from the bound port, rather than from the inner network" export:"true"`
 	SwarmMode             bool             `description:"Use Docker on Swarm Mode" export:"true"`
+	DefaultNetwork        string           `description:"Default Docker network used" export:"true"`
```
Comment (ldez) [inline provider/docker/docker.go]: To be homogeneous with `domain`, we need to keep `Network`.

Comment (emilevauge) [reply provider/docker/docker.go]: :grin:

Comment (jbdoumenjou) [reply provider/docker/docker.go]: Are you both ok with Network ? :)

File: docs/configuration/backends/docker.md (hunk 1 above)
Comment (jbdoumenjou) [reply docs/configuration/backends/docker.md]: done

**Pull request merged**